src/Cache/Signature.php

```diff
@@ -73,4 +73,19 @@ public function equals(SignatureInterface $signature)
 
         return true;
     }
+
+    private static function utf8Encode(array $data)
+    {
+        if (!function_exists('utf8_encode') || !function_exists('mb_detect_encoding')) {
+            return $data;
+        }
+
+        array_walk_recursive($data, function (&$item) {
+            if (is_string($item) && !mb_detect_encoding($item, 'utf-8', true)) {
```

Comment:
The problem with the `utf8_encode` function is that it might produce garbled output when the input is already UTF8 encoded. However it should still be an UTF8 encoded string so could be used as "signature part" to check against.

Looking at this PR, and I try not to talk anyone down here, I wonder, for the cache we need a signature so we can test if it is still valid. Can we do this without introducing the new dependencies? For example, if we serialize the configuration and MD5 (or other hashing, security is not an issue here) that, can we use that as way of comparing if the cache is invalid? (I know it is slower than json encoding, but it would be without new dependencies and we "only" need to do it one time for each run).

Besides that, I think if this PR solves an issue the current way, than we should add mb_string and xml as dependencies, otherwise the issue still remains for users without it. Next the the polyfil for 5.5 can go IMHO and replaced by detecting if the function is available. The details it provides are nice and OK, but not worth the additional depend. as users prop. won't use the additional information to fix the cache file.